x11rb-protocol/src/connection/mod.rs

```diff
@@ -238,10 +244,29 @@ impl Connection {
             }
         } else {
             // It is an event
-            self.pending_events.push_back((seqno, packet));
+            self.enqueue_event(seqno, packet);
         }
     }
 
+    /// Determine whether an event should be enqueued in the special event queue
+    /// or the normal event queue.
+    fn enqueue_event(&mut self, seqno: u64, packet: Vec<u8>) {
+        // all special events are XGE events
+        if packet[0] & 0x7F == GE_GENERIC_EVENT {
+            // the EID will be between bytes 12 and 16
+            let eid = u32::from_ne_bytes([packet[12], packet[13], packet[14], packet[15]]);
```

Comment:
So the logic here is: 
- an event that has either 35 or 163 in its first byte
- and that matches some "special" value in its bytes 12-16.
- libxcb also matches the extension that caused the request

This does not even match the event type field of the GeGenericEvent!

So....
```
$ git grep xge=
xcb-proto-1.14-1-g2b3559c/doc/xml-xcb.txt:       [[no-sequence-number="true"] | [xge="true"]]>
xcb-proto-1.14-1-g2b3559c/doc/xml-xcb.txt:  <allowed extension="identifier" xge="boolean"
xcb-proto-1.14-1-g2b3559c/doc/xml-xcb.txt:  In the current implementation, only xge="false" is supported.
xcb-proto-1.14-1-g2b3559c/src/present.xml:  <event name="ConfigureNotify" number="0" xge="true">
xcb-proto-1.14-1-g2b3559c/src/present.xml:  <event name="CompleteNotify" number="1" xge="true">
xcb-proto-1.14-1-g2b3559c/src/present.xml:  <event name="IdleNotify" number="2" xge="true">
xcb-proto-1.14-1-g2b3559c/src/present.xml:  <event name="RedirectNotify" number="3" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="DeviceChanged" number="1" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="KeyPress" number="2" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="ButtonPress" number="4" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="Enter" number="7" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="Hierarchy" number="11" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="Property" number="12" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="RawKeyPress" number="13" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="RawButtonPress" number="15" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="TouchBegin" number="18" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="TouchOwnership" number="21" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="RawTouchBegin" number="22" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:    <event name="BarrierHit" number="25" xge="true">
xcb-proto-1.14-1-g2b3559c/src/xinput.xml:        <allowed extension="Input" xge="false" opcode-min="0" opcode-max="16" />
xcb-proto-1.14-1-g2b3559c/src/xproto.xml:  <event name="GeGeneric" number="35" xge="true">
```
For all the "present"-events, it matches the `event` field. Not much surprise here, this mechanism was designed for present.

For XInput, this matches...
- DeviceValuator's second valuator field (so... if I use my touchscreen in just the right way, I might accidentially match a special event? Good luck debugging that)
- BarrierHit's timestamp field `time` (same for BarrierLeave)
- KeyPress's `event` field (same for KeyRelease, ButtonPress, ButtonRelease, MotionNotify, ProximityIn, ProximityOut)
- `FocusIn` is fun. It matches the `mode` and `device_id` field and two bytes of padding. (Same for FocusOut)
- DeviceStateNotify has a `buttons: [u8; 4]` in just the right position
- DeviceMappingNotify only has padding; same for ChangeDeviceNotify, DevicePresenceNotify, DevicePropertyNotify
- for KeyStateNotify, some random keys are hit (and ButtonStateNotify: Some random buttons)
- DeviceChanged's `time` field